### PR TITLE
fix(2767): Job permutation for stage in validator

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Need to fix syntax for job permutation, should only take one stage.

## Objective

This PR changes job permutation validation field from `stages` to `stage`.

## References

Related to https://github.com/screwdriver-cd/config-parser/pull/143#discussion_r1312135136

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: api/validator.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const SCHEMA_JOB_PERMUTATION = Joi.object()
         secrets: Job.secrets,
         settings: Job.settings,
         sourcePaths: Job.sourcePaths,
-        stages: Base.stages,
+        stage: Base.stage,
```
Comment (sagar1312) [inline api/validator.js:39]: Do we need to allow the entire stage object to be stored in a job permutation?
```
const SCHEMA_STAGE = Joi.object()
    .keys({
        description: Joi.string(),
        jobs: Joi.array().items(Job.jobName).min(0),
        setup: SCHEMA_SETUP_TEARDOWN_JOB,
        teardown: SCHEMA_SETUP_TEARDOWN_JOB,
        requires: Job.requires
    })
    .unknown(false);
```

I believe we only need stage `name` (or `id`) for reverse lookup per [this change](https://github.com/screwdriver-cd/config-parser/pull/143/files#diff-650a22c636f250a0b20166054e7ffaa7a337afe42b97be42acc2728c96107d0aR564).
```suggestion
        stage: Joi.object({
            name:  Joi.string().regex(Regex.STAGE_NAME).required()
        }).keys().unknown(false),
```

**Review by sagar1312 — CHANGES_REQUESTED** [review]
Please add tests to cover this.

File: api/validator.js [hunk 2]
```diff
@@ -36,7 +36,11 @@ const SCHEMA_JOB_PERMUTATION = Joi.object()
         secrets: Job.secrets,
         settings: Job.settings,
         sourcePaths: Job.sourcePaths,
-        stages: Base.stages,
+        stage: Joi.object()
+            .keys({
+                name: Joi.string().regex(Regex.STAGE_NAME).optional()
+            })
+            .unknown(false),
```
Comment (sagar1312) [inline api/validator.js:43]: Having `stage` should be optional. If stage object exists, it must have `name`
```suggestion
        stage: Joi.object()
            .keys({
                name: Joi.string().regex(Regex.STAGE_NAME).required()
            })
            .optional()
            .unknown(false),
```

Comment (tkyi) [reply api/validator.js]: was troubleshooting something for config-parser and forgot to change it back

**Review by sagar1312 — APPROVED** [review]

**Pull request merged**